src/content/blog/astro-330.mdx

```diff
@@ -0,0 +1,120 @@
+lang: "en"
+---
+
+Astro 3.3 is out with a brand new, experimental `<Picture />` component, improved compatibility for syntax highlighting, package provenance, and a handful of other quality of life improvements.
+
+- **[New `<Picture />` component](#picture-component)**
+- **[Syntax Highlighting improvements](#syntax-highlighting)**
+- **[Package Provenance](#package-provenance)**
+- **[Quality of Life Improvements](#quality-of-life-improvements)**
+
+```
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Picture Component
+
+When we announced [Better Images in Astro](/blog/images/) with our built-in image optimization feature, we were proud to share our new vision for working with images. In 3.0 we removed our old image integration, and along with it some existing features we know you were using. We decided to build back better, with a stronger image story that was only getting started.
+
+After [a comprehensive RFC](https://github.com/withastro/roadmap/pull/715) and [the most highly upvoted PR](https://github.com/withastro/astro/pull/8620) in Astro's history, we are incredibly excited to introduce a built-in `<Picture />` component!
+
+The new, experimental `<Picture />` component can be used to generate a `<picture>` element with multiple `<source>` elements.
+
+The example below uses the `format` property to generate a `<source>` in each of the specified image formats:
+
+```astro
+---
+import { Picture } from 'astro:assets';
+import myImage from './my-image.jpg';
+---
+
+<Picture src={myImage} formats={['avif', 'webp']} alt="My super image in multiple formats!" />
+```
+
+The above code will generate the following HTML, and allow the browser to determine the best image to display:
+
+```html
+<picture>
+  <source srcset="..." type="image/avif" />
+  <source srcset="..." type="image/webp" />
+  <img src="..." alt="My super image in multiple formats!" />
+</picture>
+```
+
+The `Picture` component takes all the same props as the `Image` component, including the new `densities` and `widths` properties.
+
+## `srcset` support
+
+In addition to the `<Picture />` element, two new properties have been added to `Image` and `getImage()`: `densities` and `widths`.
+
+These properties can be used to generate a `srcset` attribute, either based on absolute widths in pixels (e.g. [300, 600, 900]) or pixel density descriptors (e.g. `["2x"]` or `[1.5, 2]`).
+
+```astro
+---
+import { Image } from 'astro';
+import myImage from './my-image.jpg';
+---
+
+<Image src={myImage} width={myImage.width / 2} densities={[1.5, 2]} alt="My cool image" />
+```
+
+```html
+<img
+  src="/_astro/my_image.hash.webp"
+  srcset="/_astro/my_image.hash.webp 1.5x, /_astro/my_image.hash.webp 2x"
+  alt="My cool image"
+/>
+```
+
+## Syntax Highlighting
+
+Internally, Astro has migrated our internal syntax highlighter library from `shiki` to `shikiji`, an ESM-focused alternative that simplifies bundling and maintenance.
```

Comment:
Small rework suggestion to remove the double “internal”:

```suggestion
We migrated Astro’s internal syntax highlighter library from `shiki` to `shikiji`, an ESM-focused alternative that simplifies bundling and maintenance.
```

Reply:
Wonderful catch, thanks!